boa_engine/src/object/operations.rs

```diff
@@ -656,19 +657,331 @@ impl JsObject {
+    ///
+    /// Ensure private elements can be added to an object.
+    ///
+    /// More information:
+    ///  - [ECMAScript specification][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-hostensurecanaddprivateelement
+    #[allow(clippy::unnecessary_wraps, clippy::unused_self)]
+    fn host_ensure_can_add_private_element(&self, _context: &mut Context<'_>) -> JsResult<()> {
+        // TODO: Make this operation host-defined.
+        Ok(())
+    }
+
+    /// Abstract operation `PrivateGet ( O, P )`
+    ///
+    /// Get the value of a private element.
+    ///
+    /// More information:
+    ///  - [ECMAScript specification][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-privateget
+    pub(crate) fn private_get(
+        &self,
+        name: &PrivateName,
+        context: &mut Context<'_>,
+    ) -> JsResult<JsValue> {
+        // 1. Let entry be PrivateElementFind(O, P).
+        let entry = self.private_element_find(name, true, true);
+
+        match &entry {
+            // 2. If entry is empty, throw a TypeError exception.
+            None => Err(JsNativeError::typ()
+                .with_message("Private element does not exist on object")
+                .into()),
+
+            // 3. If entry.[[Kind]] is field or method, then
+            // a. Return entry.[[Value]].
+            Some(PrivateElement::Field(value)) => Ok(value.clone()),
+            Some(PrivateElement::Method(value)) => Ok(value.clone().into()),
+
+            // 4. Assert: entry.[[Kind]] is accessor.
+            Some(PrivateElement::Accessor { getter, .. }) => {
+                // 5. If entry.[[Get]] is undefined, throw a TypeError exception.
+                // 6. Let getter be entry.[[Get]].
+                let getter = if let Some(getter) = getter {
+                    getter
+                } else {
+                    return Err(JsNativeError::typ()
+                        .with_message("private property was defined without a getter")
+                        .into());
+                };
+
+                // 7. Return ? Call(getter, O).
+                getter.call(&self.clone().into(), &[], context)
+            }
+        }
+    }
+
+    /// Abstract operation `PrivateSet ( O, P, value )`
+    ///
+    /// Set the value of a private element.
+    ///
+    /// More information:
+    ///  - [ECMAScript specification][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-privateset
+    pub(crate) fn private_set(
+        &self,
+        name: &PrivateName,
+        value: JsValue,
+        context: &mut Context<'_>,
+    ) -> JsResult<()> {
+        // 1. Let entry be PrivateElementFind(O, P).
+        // Note: This function is inlined here for mutable access.
+        let mut object_mut = self.borrow_mut();
+        let mut entry = None;
+        for (key, value) in &mut object_mut.private_elements {
+            if key == name {
+                entry = Some(value);
+            }
+        }
```

Comment:
Maybe something like
```suggestion
        let mut entry = object_mut.private_elements.iter_mut().find(|(key, _)| key == name);
```